### ISSUE TITLE
[Bug] [Module Name] Rest API：hazelcast/rest/maps/submit-job return an wrong jobId

### ISSUE THREAD

**Issue by chenliejun** [apache/seatunnel#7146]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

I send the post api as following parameter format，got the result：
{
  "jobId": 862976196311253000,
  "jobName": "event_test"
}

but I use terminal command $SEATUNNEL_HOME/bin/seatunnel.sh -l got another jobId：862969647010611201 ，and I confirmed the 862969647010611201 is correct because I can stop it by stop job rest api。

I restart the cluster master Server serveral times got the same error ,the bug seems like [Issue 5421](https://github.com/apache/seatunnel/issues/5241), but it closed for it still occur,so I n

### SeaTunnel Version

2.3.5

### SeaTunnel Config

```conf
seatunnel:
  engine:
    history-job-expire-minutes: 1440
    backup-count: 1
    queue-type: blockingqueue
    print-execution-info-interval: 60
    print-job-metrics-info-interval: 60
    slot-service:
      dynamic-slot: true
    event-report-http:
      url: "http://xx.xx.xx.xx:1024/event/report"
      headers:
        Content-Type: application/json
    checkpoint:
      interval: 10000
      timeout: 60000
      storage:
        type: hdfs
        max-retained: 3
        plugin-config:
          storage.type: hdfs
          fs.defaultFS: file:///data/seatunnel/checkpoint_snapshot/
```

### Running Command

```shell
POST /hazelcast/rest/maps/submit-job HTTP/1.1
Host: xx.xx.xx.xx:5801

{
    "env": {
        "job.mode": "batch",
        "job.name": "event_test"
    },
    "source": [
        {
            "plugin_name": "FakeSource",
            "result_table_name": "fake",
            "row.num": 100,
            "schema": {
                "fields": {
                    "name": "string",
                    "age": "int",
                    "card": "int"
                }
            }
        }
    ],
    "transform": [
    ],
    "sink": [
        {
            "plugin_name": "Console",
            "source_table_name": ["fake"]
        }
    ]
}
```

### Error Exception

```log
the rest api return information was different from the command monitor metrics。
```

### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

java v1.8

### Screenshots

![1720516324419](https://github.com/apache/seatunnel/assets/7288050/c85dac3b-29c9-4c59-8954-8566125684fb)
![image](https://github.com/apache/seatunnel/assets/7288050/d9f3e725-a7ae-45c1-9bd0-803caf764f66)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Hisoka-X** [comment]
cc @liugddx as well.

**Comment by Hisoka-X** [comment]
We should fix this before 2.3.6.  cc @hailin0 @liugddx @EricJoy2048

**Comment by liugddx** [comment]
> We should fix this before 2.3.6. cc @hailin0 @liugddx @EricJoy2048

👀

**Comment by liugddx** [comment]
Our test cases already cover this scenario, Have we fixed this issue in 2.3.6? 
https://github.com/apache/seatunnel/blob/57e5627e89e870f26c1b0774cc225eaabe7f3c9a/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterSeaTunnelContainer.java#L443-L445
